tests/job_enable_common.erl

```diff
@@ -0,0 +1,572 @@
+        false ->
+            ?assertEqual({error, ?ERRMSG_BIN(Class)}, Result)
+    end;
+
+% protobuf list-keys only does streams, so skip the non-stream test
+test_operation(_, ?TOKEN_LIST_KEYS, _, pbc) ->
+    ok;
+test_operation(Node, ?TOKEN_LIST_KEYS = Class, Enabled, http = Scheme) ->
+    URL = make_url(Node, Scheme, ["/buckets/",
+        erlang:binary_to_list(populated_bucket()), "/keys?keys=true"]),
+    Result = ibrowse:send_req(URL, [], get, [], [{response_format, binary}]),
+    ?assertMatch({ok, _, _, _}, Result),
+    {_, Code, _, Body} = Result,
+    case Enabled of
+        true ->
+            {struct, PList} = mochijson2:decode(
+                unicode:characters_to_list(Body, utf8)),
+            Keys = proplists:get_value(<<"keys">>, PList, []),
+            ?assertEqual({"200", test_keys()}, {Code, lists:sort(Keys)});
+        false ->
+            ?assertEqual({"403", ?ERRMSG_BIN(Class)}, {Code, Body})
+    end;
+
+test_operation(Node, ?TOKEN_LIST_KEYS_S = Class, false, http = Scheme) ->
+    URL = make_url(Node, Scheme, ["/buckets/",
+        erlang:binary_to_list(populated_bucket()), "/keys?keys=stream"]),
+    Result = ibrowse:send_req(URL, [], get),
+    ?assertMatch({ok, _, _, _}, Result),
+    {_, Code, _, Body} = Result,
+    ?assertEqual({"403", ?ERRMSG_TXT(Class)}, {Code, Body});
+test_operation(Node, ?TOKEN_LIST_KEYS_S = Class, Enabled, ClientType) ->
+    {_, Mod, Conn} = Client = open_client(ClientType, Node),
+    % 'rhc' and 'riakc_pb_socket' list_keys always use stream_list_keys
+    Result = Mod:list_keys(Conn, populated_bucket()),
+    close_client(Client),
+    case Enabled of
+        true ->
+            ?assertMatch({ok, _}, Result),
+            {ok, Keys} = Result,
+            ?assertEqual(test_keys(), lists:sort(Keys));
+        false ->
+            ?assertEqual({error, ?ERRMSG_BIN(Class)}, Result)
+    end;
+
+test_operation(Node, ?TOKEN_MAP_REDUCE = Class, Enabled, ClientType) ->
+    Bucket = populated_bucket(),
+    {_, Mod, Conn} = Client = open_client(ClientType, Node),
+    Result = Mod:mapred(Conn, Bucket, []),
+    close_client(Client),
+    case Enabled of
+        true ->
+            ?assertMatch({ok, [{_, _}]}, Result),
+            {ok, [{_, Pairs}]} = Result,
+            Expect = case ClientType of
+                pbc ->
+                    [{Bucket, Key} || Key <- test_keys()];
+                http ->
+                    [[Bucket, Key] || Key <- test_keys()]
+            end,
+            ?assertEqual(Expect, lists:sort(Pairs));
+        false ->
+            case ClientType of
+                pbc ->
+                    ?assertEqual({error, ?ERRMSG_BIN(Class)}, Result);
+                http ->
+                    ?assertMatch({error, {"403", _}}, Result)
+            end
+    end;
+
+test_operation(Node, ?TOKEN_SEC_INDEX = Class, Enabled, pbc = ClientType) ->
+    Bucket = populated_bucket(),
+    Index = index_2i(),
+    Num = random:uniform(num_keys()),
```

Comment:
I'm not sure if this ultimately really matters or not, but it looks like we're never seeding the `random` module. This means it will automatically call `random:seed/0` the first time we call `random:uniform`, which uses a fixed value for the seed, which means we'll get the same sequence of "random" numbers every time we run a test.